bin/ocamlformat-rpc/index.mld

```diff
@@ -0,0 +1,42 @@
+{0 ocamlformat-rpc}
+
+[ocamlformat-rpc] is a RPC mode for OCamlFormat, a tool to format OCaml code.
+
+[ocamlformat-rpc] listens to RPC requests, provided on the standard input, and prints the response on the standard output.
+
+{1 Agreeing on a RPC version}
+
+Before the client and the server agree on a common version to use the following commands are available:
+- [Halt] to close the connection to the RPC. The command is encoded as the canonical s-expression [Atom "Halt"];
+- [Version <v>] to ask the server to use version [v]. The command expects a canonical s-expression of the form [List [Atom "Version"; Atom v]] where [v] is of the version name (e.g. ["v1"]).
+
+If the server agrees upon the version he will send the reply [Version v] and the protocol version is set to [v], to use another version later the client has to close the connexion and start a new one. If the server cannot use version [v] he might propose another version [w] by sending the reply [Version w] that the client can accept by sending the same request for version [w], or propose another version. If the server cannot propose another version it will close the connection. Unknown commands are ignored.
+
+Once the client and the server agree on a common version, the requests you can send may differ from one version to another.
+
+{1 Commands}
+
+All versions support the following commands:
+- [Halt] to end the communication with the RPC server. The caller must close the input and output channels.
```

Comment:
I think this section isn't useful, it could be copied into both sections below. Also, something tells me this command is useless if it waits for the input to be closed.

Reply:
done